Firestore/src/Query.php

```diff
@@ -1014,4 +978,144 @@ private function basePath()
+                $value = array_map(function ($value) use ($basePath) {
+                    return $this->createDocumentReference($basePath, $value);
+                }, (array) $value);
+            } else {
+                $value = $this->createDocumentReference($basePath, $value);
+            }
+        }
+
+        if ((is_float($value) && is_nan($value)) || is_null($value)) {
+            if ($operator !== FieldFilterOperator::EQUAL) {
+                throw new \InvalidArgumentException('Null and NaN are allowed only with operator EQUALS.');
+            }
+
+            $unaryOperator = is_nan($value)
+                ? self::OP_NAN
+                : self::OP_NULL;
+
+            $filter = [
+                'unaryFilter' => [
+                    'field' => [
+                        'fieldPath' => $escapedPathString
+                    ],
+                    'op' => $unaryOperator
+                ]
+            ];
+        } else {
+            $encodedValue = $operator === FieldFilterOperator::IN
+                ? $this->valueMapper->encodeMultiValue((array)$value)
+                : $this->valueMapper->encodeValue($value);
+
+            $filter = [
+                'fieldFilter' => [
+                    'field' => [
+                        'fieldPath' => $escapedPathString,
+                    ],
+                    'op' => $operator,
+                    'value' => $encodedValue
+                ]
+            ];
+        }
+        
+        return $filter;
+    }
+
+    /**
+     * Find and set the the filter type to either Field or Unary.
+     *
+     * @param array $filter A field/unary filter array.
+     * @return array A non composite Filter array.
+     * @throws \InvalidArgumentException If an invalid filter is encountered.
+     */
+    private function findAndSetNonCompositeFilter($filter)
+    {
+        $nonCompositeFilter = $this->createNonCompositeFilter(
+            $filter['fieldFilter']['field'],
+            $filter['fieldFilter']['op'],
+            $filter['fieldFilter']['value']
+        );
+
+        return $nonCompositeFilter;
+    }
+
+    /**
+     * Create a composite filter from a filter array.
+     *
+     * @param array $filters A non sanitized composite filter array.
+     * @return array A sanitized composite Filter array.
+     * @throws \InvalidArgumentException If an invalid filter is encountered.
+     */
+    private function createCompositeFilter($filters)
+    {
+        foreach ($filters as $key => $filter) {
+            if ($key === 'fieldFilter') {
+                $nonCompositeFilter = $this->findAndSetNonCompositeFilter($filters);
+                if (isset($nonCompositeFilter['unaryFilter'])) {
+                    unset($filters['fieldFilter']);
+                    $filters['unaryFilter'] = $nonCompositeFilter['unaryFilter'];
+                } else {
+                    $filters['fieldFilter'] = $nonCompositeFilter['fieldFilter'];
```

Comment:
dont you want to `unset($filters['unaryFilter'])` here?

Reply:
Nop, Filter class always set the given non composite filter as `fieldFilter`. So, if the given filter is `unaryFilter` we need to unset the `fieldFilter` else we just need to replace the existing value for the `fieldFilter`. No unset is required. Hope it is clear now.